src/solvers/dg/dg_2d.jl

```diff
@@ -8,7 +8,7 @@
 function create_cache(mesh::TreeMesh{2}, equations::AbstractEquations{2},
```

Comment:
Should this be specific to the serial tree?
```suggestion
function create_cache(mesh::SerialTreeMesh{2}, equations::AbstractEquations{2},
```

Reply:
It's supposed to be a general fallback. Note that
```julia
julia> TreeMesh{2}
TreeMesh{2,TreeType} where TreeType<:Trixi.AbstractTree{2}
```
Hence, we can use `TreeMesh{2}` as general fallback without changing the serial code at all and create specializations using dispatch on the parallel tree mesh.

Reply:
:+1: